### PR TITLE
Cortx 32237 f 33 a 2

### PR DESCRIPTION
**Pull request by archanabpatil**
Signed-off-by: archanabpatil <66412855+archanabpatil@users.noreply.github.com>

# Problem Statement
- RBF for Multipart versioning tests 1-4

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [x] Attach test execution logs
-  [x] Collection tested and no collection error introduced (`pytest --local True --collect-only`)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: libs/s3/s3_versioning_common_test_lib.py [hunk 1]
```diff
@@ -157,7 +157,6 @@ def check_list_object_versions(s3_ver_test_obj: S3VersioningTestLib,
     :keyword "expected_error": Error message string to verify in case, error is expected
     :keyword "list_params": Dictionary of query parameters to pass List Object Versions call
     """
-    LOG.info("Fetching bucket object versions list")
```
Comment (archanabpatil) [inline libs/s3/s3_versioning_common_test_lib.py:160]: Removed this line as it was getting twice : here and in list_object_versions() of  s3_versioning_test_lib.py

**Comment by archanabpatil** [comment]
TE: https://jts.seagate.com/browse/TEST-43543 : 
Tests: TEST-41284,   TEST-41285,  TEST-41286,  TEST-41287

File: tests/s3/versioning/test_versioning_multipart.py [hunk 2]
```diff
@@ -124,9 +129,18 @@ def test_mpu_ver_enabled_bkt_41285(self):
...
                                      versions_dict=versions, is_multipart=True,
                                      total_parts=2, file_size=10, is_unversioned=False)
+        parts = get_precalculated_parts(self.test_file_path, mp_config["part_sizes"],
+                                        chunk_size=mp_config["chunk_size"])
+        s3ver_cmn_lib.upload_version(self.s3mp_test_obj, self.bucket_name, self.object_name,
+                                     self.test_file_path, versions_dict=versions, parts=parts,
+                                     is_mpu_with_lst_mpu=True)
+        self.log.info("Verions has etag and version id %s", versions)
         self.log.info("Step 8: List Object Versions")
+        params = {"Prefix": "", "MaxKeys": 1000, "EncodingType": "url",
```
Comment (apoorva-rao) [inline tests/s3/versioning/test_versioning_multipart.py:139]: You would need to call check_list_object_versions here without any list_params. List response should return expected versions without needing KeyMarker specified

**Review by apoorva-rao — CHANGES_REQUESTED** [review]
@archanabpatil Please see the inline review comments

File: libs/s3/s3_versioning_common_test_lib.py [hunk 3]
```diff
@@ -411,11 +410,25 @@ def upload_version(s3_test_obj: Union[S3TestLib, S3MultipartTestLib], bucket_nam
...
+    calc_parts = kwargs.get("parts", "None")
     if is_multipart:
         res = s3_test_obj.complete_multipart_upload_with_di(
             bucket_name, object_name, file_path, total_parts=total_parts, file_size=file_size)
+    elif is_mpu_with_list_mpu:
+        mpu_id, _, parts = initiate_upload_list_mpu(s3_test_obj, bucket_name, object_name,
+                                                    calc_parts)
+        res = s3_test_obj.complete_multipart_upload(mpu_id, parts, bucket_name, object_name)
+        upload_etag = res[1]["ETag"]
+        LOG.info("Get the uploaded object")
```
[the remark this answers is not in the document]
Comment (archanabpatil) [reply libs/s3/s3_versioning_common_test_lib.py:425]: I agree. I will remove line 425-431. Anyways in each test we are doing check_get_head_object_version , so just avoiding repeating that as well

File: tests/s3/versioning/test_versioning_multipart.py (hunk 2 above)
Comment (archanabpatil) [reply tests/s3/versioning/test_versioning_multipart.py:139]: okay,

**Comment by archanabpatil** [comment]
https://jts.seagate.com/browse/TEST-43871: TE: 7 tests passed as expected. 3 are failed as its being worked on in other PR by @apoorva-rao .

File: libs/s3/s3_versioning_common_test_lib.py [hunk 4]
```diff
@@ -731,29 +735,26 @@ def delete_object_tagging(s3_tag_test_obj: S3TaggingTestLib, s3_ver_test_obj: S3
...
-        sorted_lst = sorted(parts_details, key=lambda x: x['PartNumber'])
-        res = self.s3_mp_test_obj.list_parts(mpu_id, bucket_name, object_name)
-        assert_utils.assert_true(res[0], res[1])
-        assert_utils.assert_not_in("VersionId", resp[1])
-        self.log.info("List Multipart uploads")
-        return mpu_id, resp, sorted_lst
-    return mpu_id
+    LOG.info("Uploading parts")
+    resp = s3_mp_test_obj.upload_parts_parallel(mpu_id, bucket_name,
+                                                      object_name, parts=parts_)
```
Comment (apoorva-rao) [inline libs/s3/s3_versioning_common_test_lib.py:749]: Let's correct indentation here

**Review by apoorva-rao — CHANGES_REQUESTED** [review]
@archanabpatil minor comment added inline, rest looks good

File: libs/s3/s3_versioning_common_test_lib.py (hunk 4 above)
Comment (archanabpatil) [reply libs/s3/s3_versioning_common_test_lib.py]: done

**Review by DhobaleVishal — APPROVED** [review]

**Review by VikasKumarC — APPROVED** [review]

**Review by ravindrachoudhari — APPROVED** [review]

**Pull request merged**